### PR TITLE
add a system variable tidb_lock_unchanged_keys

### PR DESCRIPTION
**Pull request by ekexium**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/44598

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@ekexium Could you please invite a tech reviewer? Thanks!

**Comment by ekexium** [comment]
@Oreoxmt Let's wait until the TiDB PR is merged.

**Review by zyguan — APPROVED** [review]

**Comment by ekexium** [comment]
PTAL @Oreoxmt

File: system-variables.md
Comment (Oreoxmt) [inline system-variables.md:2454]:
```suggestion
### `tidb_lock_unchanged_keys` <span class="version-mark">从 v7.1.1 版本开始引入</span>
```

File: system-variables.md
Comment (Oreoxmt) [inline system-variables.md:2463]:
```suggestion
- 该变量控制在以下场景是否对某些 key 加锁。设置为 `ON` 时，都加锁，设置为 `OFF` 时，都不加锁。
    - 在 `INSERT IGNORE` 语句和 `REPLACE` 语句中值重复的 key。在 v6.1.6 之前版本中，这些 key 不加锁。这个问题已在 [#42121](https://github.com/pingcap/tidb/issues/42121) 修复。
    - 在 `UPDATE` 语句中值没有改变的唯一索引 key。在 v6.5.2 之前版本中，这些 key 不加锁。这个问题已在 [#36438](https://github.com/pingcap/tidb/issues/36438) 修复。
- 为保证事务行为的一致性和合理性，不推荐修改该值。当升级 TiDB 后因为这两项修复导致严重的性能问题，且可以接受不加锁的行为（见上述 issue）时，可以将该变量设置为 `OFF`。
```

File: system-variables.md
Comment (ekexium) [reply system-variables.md:2454]: That's inaccurate. It's not in 7.2.0.

Comment (Oreoxmt) [reply system-variables.md:2454]: The label "New in v7.1.1" indicates that this variable was introduced after v7.1.1. Given that v7.2.0 was released before v7.1.1, (see [release timeline](https://docs.pingcap.com/tidb/dev/release-timeline)), this variable doesn't exist in v7.2.0.

To avoid any confusion about whether this variable is available in v7.2.0, we can switch the base of this PR to release-7.1 and only add this variable to the v7.1 documentation.

After the code freeze of v7.3.0, we can cherry-pick this PR to the master branch and update the description to "New in v7.1.1 and v7.3.0".

Comment (Oreoxmt) [reply system-variables.md:2454]: @ekexium PTAL

Comment (Oreoxmt) [reply system-variables.md:2454]: @ekexium I have created the PR https://github.com/pingcap/docs-cn/pull/14508 and https://github.com/pingcap/docs/pull/14222 for v7.1.1, PTAL

File: system-variables.md
Comment (Oreoxmt) [inline system-variables.md:2463]:
```suggestion
### `tidb_lock_unchanged_keys` <span class="version-mark">从 v7.1.1 和 v7.3.0 版本开始引入</span>

- 作用域：SESSION | GLOBAL
- 是否持久化到集群：是
- 类型：布尔型
- 默认值：`ON`
- 该变量控制在以下场景是否对某些 key 加锁。设置为 `ON` 时，都加锁，设置为 `OFF` 时，都不加锁。
    - 在 `INSERT IGNORE` 语句和 `REPLACE` 语句中值重复的 key。在 v6.1.6 之前版本中，这些 key 不加锁。这个问题已在 [#42121](https://github.com/pingcap/tidb/issues/42121) 修复。
    - 在 `UPDATE` 语句中值没有改变的唯一索引 key。在 v6.5.2 之前版本中，这些 key 不加锁。这个问题已在 [#36438](https://github.com/pingcap/tidb/issues/36438) 修复。
- 为保证事务行为的一致性和合理性，不推荐修改该值。如果在升级 TiDB 后因为这两项修复导致严重的性能问题，且可以接受不加锁的行为（见上述 Issue），可以将该变量设置为 `OFF`。
```

**Review by Oreoxmt — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**
